### PR TITLE
feat: filter inputs - week picker in mantine

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: #7313 

### Description:

tested with:

- [x] updated project settings "start of week"
- [x] us/east timezone
- [x] uk timezone
- [x] japan timezone

CleanShot 2023-10-16 at 20 02 15@2x

testing video (⚠️ something fails in the end and in the second video I demonstrated the fix):

https://github.com/lightdash/lightdash/assets/962095/6037b49d-431e-4ef0-8a6b-dd62311e157f

https://github.com/lightdash/lightdash/assets/962095/f2500ab4-c443-4b1d-a213-4f8f875ab0af

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Comment by IrakliJani** [comment]
remaining items:
- [ ] fix e2e test issue (selector)

File: packages/frontend/src/components/common/Filters/FilterInputs/dateUtils.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import { WeekDay } from '@lightdash/common';
+import { DayOfWeek } from '@mantine/dates';
+import dayjs from 'dayjs';
+
+//
+// date inputs accepts start of the week day
+// prop is firstDayOfWeek: number 0-6
+// 0 – Sunday, 6 – Saturday, defaults to 1 – Monday
+// our internal WeekDay enum is a range from 0 (Monday) to 6 (Sunday)
```
Comment (magnew) [inline packages/frontend/src/components/common/Filters/FilterInputs/dateUtils.ts:9]: Wow, too bad we did that 😄. Cron has sunday as 0

File: packages/frontend/src/components/common/Filters/FilterInputs/dateUtils.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+//
+export const convertWeekDayToDayOfWeek = (weekDay: WeekDay): DayOfWeek => {
```
Comment (magnew) [inline packages/frontend/src/components/common/Filters/FilterInputs/dateUtils.ts:11]: This is small, but the name of this function could be easier to understand. Right now I have no idea what it does without reading it. 

Maybe normalizeWeekDayIndex? Or something like that.

**Review by magnew — APPROVED** [review]
I tested this and it looks great! Nice work. Code is very nice as well. I left a non-blocking readability suggestion.

File: packages/frontend/src/components/common/Filters/FilterInputs/dateUtils.ts (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/components/common/Filters/FilterInputs/dateUtils.ts]: great suggestion. called it `normalizeWeekDay` 👌

**Pull request merged**